### PR TITLE
Upgrade the project system to depend on the latest version of Roslyn

### PR DESCRIPTION
**Pull request by jmarolf**
This updated to [this](https://github.com/dotnet/roslyn/commit/170d0bb79e28aaabc753458b9f1d0bd4b6111689) commit of roslyn.  This forces us to bring along several other VS dependencis.

NOTE: Async package made a breaking change forcing 3 tests to be disabled

### CODE REVIEW

**Comment by jmarolf** [comment]
@genlu

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS.UnitTests/Packaging/CSharpProjectSystemPackageTests.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace Microsoft.VisualStudio.Packaging
 {
     public class CSharpProjectSystemPackageTests
     {
-        [Fact]
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS.UnitTests/Packaging/CSharpProjectSystemPackageTests.cs]: Just delete them.

**Comment by davkean** [comment]
Has this been inserted into Preview 2, if so, which build? If not, we can't merge this.

**Comment by genlu** [comment]
@davkean I upgraded Roslyn to a build contains the new `SetProperty` API. But we haven't inserted anything to preview 2 yet, so this PR still can't be merged. Also, `IWorkspaceProjectContext` can be accessed from background thread now, could you please check if `ForegroundWorkspaceProjectContext` is still needed?

**Comment by davkean** [comment]
ForegroundWorkspaceProjectContext is known a issue in master, and you can ignore it. It's already been fixed.

**Comment by jmarolf** [comment]
NOTE: this PR https://github.com/dotnet/project-system/pull/4318 also depends on this getting merged.

**Comment by davidwengier** [comment]
Actually this Roslyn package isn't new enough to contain my changes, and I also need a new CPS package for that PR, so was going to tackle the integration as the last step.

**Comment by jmarolf** [comment]
> Actually this Roslyn package isn't new enough to contain my changes

Fair enough.  Once Roslyn insertion into Preview 2 we can take literally the newest package they have in master so that's not a problem. Is the CPS package that we would need to consume published?

**Comment by davidwengier** [comment]
Yep, it is (>= `16.0.429-pre-gf4f06355f6`) but I think Drew still needs a new one at some point, which would be after my changes there, so whoever gets there first wins with that one I guess.

**Comment by drewnoakes** [comment]
The CPS change we need has been inserted to d16.0stg and a new package is on MyGet.

https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-environment-logs&releaseId=201112&environmentId=805874

**Comment by genlu** [comment]
We finally completed the first roslyn insertion for preview 2.
https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/152224?_a=files

Could you update roslyn to 2.11.0-beta2-63529-05 to match the version and merge this?

**Comment by jmarolf** [comment]
@dotnet/project-system 	this is safe to merge now as the equivalent roslyn bits are in d16.0stg.  Once things are green I am going to merge this.

**Pull request merged**